<?xml version="1.0"?>
<launch>
    <rosparam file="$(find fence_following)/config/uav.yaml" />

    <!-- MAVROS -->
    <include file="$(find mavros)/launch/px4.launch">
        <arg name="fcu_url" value="$(arg fcu_url)"/>
        <arg name="gcs_url" value=""/>
        <arg name="tgt_system" value="$(arg ID)"/>
        <arg name="tgt_component" value="1"/>
    </include>

    <!-- Fence detection -->
    <node name="fence_detector" pkg="fence_following" type="pointCloudSampler.py"/>
    <node name="fence_detector" pkg="fence_following" type="fenceDetector.py"/>

    <!-- Control -->
    <node name="state_machine" pkg="fence_following" type="stateMachine.py"/>
</launch>
